File: src/Entity/Agriculteur.php
<?php

namespace App\Entity;

use App\Repository\AgriculteurRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: AgriculteurRepository::class)]
class Agriculteur
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(inversedBy: 'agriid')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Utilisateur $utilisateur_id = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $adresseferme = null;

    #[ORM\Column(type: Types::DECIMAL, precision: 10, scale: 2, nullable: true)]
    private ?string $superficieferme = null;

    #[ORM\Column(length: 100, nullable: true)]
    private ?string $typeculture = null;

    #[ORM\Column(length: 50, nullable: true)]
    private ?string $statuscompte = null;

    #[ORM\Column(nullable: true)]
    private ?bool $compteverifie = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function setId(int $id): static
    {
        $this->id = $id;

        return $this;
    }

    public function getUtilisateurId(): ?Utilisateur
    {
        return $this->utilisateur_id;
    }

    public function setUtilisateurId(?Utilisateur $utilisateur_id): static
    {
        $this->utilisateur_id = $utilisateur_id;

        return $this;
    }

    public function getAdresseferme(): ?string
    {
        return $this->adresseferme;
    }

    public function setAdresseferme(?string $adresseferme): static
    {
        $this->adresseferme = $adresseferme;

        return $this;
    }

    public function getSuperficieferme(): ?string
    {
        return $this->superficieferme;
    }

    public function setSuperficieferme(?string $superficieferme): static
    {
        $this->superficieferme = $superficieferme;

        return $this;
    }

    public function getTypeculture(): ?string
    {
        return $this->typeculture;
    }

    public function setTypeculture(?string $typeculture): static
    {
        $this->typeculture = $typeculture;

        return $this;
    }

    public function getStatuscompte(): ?string
    {
        return $this->statuscompte;
    }

    public function setStatuscompte(?string $statuscompte): static
    {
        $this->statuscompte = $statuscompte;

        return $this;
    }

    public function isCompteverifie(): ?bool
    {
        return $this->compteverifie;
    }

    public function setCompteverifie(?bool $compteverifie): static
    {
        $this->compteverifie = $compteverifie;

        return $this;
    }
}

File: src/main/java/com/agrifund/entities/Agriculteur.java
package com.agrifund.entities;

import java.math.BigDecimal;

/**
 * Entité Agriculteur - hérite de Utilisateur
 * Correspond à la table Agriculteur dans la base de données
 */
public class Agriculteur extends Utilisateur {

    private int agriculteurId; // ID dans la table Agriculteur
    private int utilisateurId; // Référence vers la table Utilisateur
    private String adresseFerme;
    private BigDecimal superficieFerme;
    private String typeCulture;
    private String statusCompte;
    private boolean compteVerifie;

    // Constructeurs
    public Agriculteur() {
        super();
        this.statusCompte = "en_attente";
        this.compteVerifie = false;
    }

    public Agriculteur(String nom, String prenom, String email, String password,
                       String adresseFerme, BigDecimal superficieFerme, String typeCulture) {
        super(nom, prenom, email, password);
        this.adresseFerme = adresseFerme;
        this.superficieFerme = superficieFerme;
        this.typeCulture = typeCulture;
        this.statusCompte = "en_attente";
        this.compteVerifie = false;
    }

    // Getters et Setters
    public int getAgriculteurId() {
        return agriculteurId;
    }

    public void setAgriculteurId(int agriculteurId) {
        this.agriculteurId = agriculteurId;
    }

    public int getUtilisateurId() {
        return utilisateurId;
    }

    public void setUtilisateurId(int utilisateurId) {
        this.utilisateurId = utilisateurId;
    }

    public String getAdresseFerme() {
        return adresseFerme;
    }

    public void setAdresseFerme(String adresseFerme) {
        this.adresseFerme = adresseFerme;
    }

    public BigDecimal getSuperficieFerme() {
        return superficieFerme;
    }

    public void setSuperficieFerme(BigDecimal superficieFerme) {
        this.superficieFerme = superficieFerme;
    }

    public String getTypeCulture() {
        return typeCulture;
    }

    public void setTypeCulture(String typeCulture) {
        this.typeCulture = typeCulture;
    }

    public String getStatusCompte() {
        return statusCompte;
    }

    public void setStatusCompte(String statusCompte) {
        this.statusCompte = statusCompte;
    }

    public boolean isCompteVerifie() {
        return compteVerifie;
    }

    public void setCompteVerifie(boolean compteVerifie) {
        this.compteVerifie = compteVerifie;
    }

    @Override
    public String toString() {
        return "Agriculteur{" +
                "agriculteurId=" + agriculteurId +
                ", adresseFerme='" + adresseFerme + '\'' +
                ", superficieFerme=" + superficieFerme +
                ", typeCulture='" + typeCulture + '\'' +
                ", statusCompte='" + statusCompte + '\'' +
                ", compteVerifie=" + compteVerifie +
                ", " + super.toString() +
                '}';
    }
}